clash-lib/src/Clash/Netlist/BlackBox.hs

```diff
@@ -564,6 +564,26 @@ mkPrimitive bbEParen bbEasD declType dst pInfo args tickDecls =
                     return (DataTag scrutHTy (Right tmpRhs),netDecl ++ scrutDecls)
               _ -> error $ $(curLoc) ++ "dataToTag: " ++ show (map (either showPpr showPpr) args)
 
+          | pNm `elem`
+            ["GHC.Prim.dataToTagSmall#", "GHC.Prim.dataToTagLarge#"] -> case args of
+              [Right _, Right _,Left (Data dc)] -> do
+                iw <- Lens.view intWidth
+                return (N.Literal (Just (Signed iw,iw)) (NumLit $ toInteger $ dcTag dc - 1),[])
+              [Right _, Right _,Left scrut] -> do
+                tcm      <- Lens.view tcCache
+                let scrutTy = inferCoreTypeOf tcm scrut
+                scrutHTy <- unsafeCoreTypeToHWTypeM' $(curLoc) scrutTy
+                (scrutExpr,scrutDecls) <-
+                  mkExpr False Concurrent (NetlistId (Id.unsafeMake "c$dtt_rhs") scrutTy) scrut
```

Comment:
I copied what was there from from `dataToTag#` and didn’t give it a second thought really.

Reply:
Ah ok, fair enough. Could you add an issue saying we should investigate this? In case someone gets an error with this ID it should point them to something tangible.

Reply:
Tracked by https://github.com/clash-lang/clash-compiler/issues/2775